ethcontract-common/src/artifact/hardhat.rs

```diff
@@ -0,0 +1,290 @@
+        }
+    }
+
+    /// Create a new hardhat loader and set an override for artifact's origins.
+    pub fn with_origin(origin: impl Into<String>) -> Self {
+        HardHatLoader {
+            origin: Some(origin.into()),
+            networks_deny_list: Vec::new(),
+            networks_allow_list: Vec::new(),
+        }
+    }
+
+    /// Set new override for artifact's origin. See [`origin`] for more info.
+    ///
+    /// [`origin`]: #structfield.origin
+    #[inline]
+    pub fn origin(mut self, origin: Option<String>) -> Self {
+        self.origin = origin;
+        self
+    }
+
+    /// Add network name or id to the list of [`allowed networks`].
+    ///
+    /// [`allowed networks`]: #structfield.networks_allow_list
+    #[inline]
+    pub fn allow_network(mut self, network: impl Into<String>) -> Self {
+        self.networks_allow_list.push(network.into());
+        self
+    }
+
+    /// Add network name or id to the list of [`denied networks`].
+    ///
+    /// [`denied networks`]: #structfield.networks_deny_list
+    #[inline]
+    pub fn deny_network(mut self, network: impl Into<String>) -> Self {
+        self.networks_deny_list.push(network.into());
+        self
+    }
+
+    /// Parse a hardhat artifact from JSON string.
+    pub fn load_from_string(&self, format: Format, json: &str) -> Result<Artifact, ArtifactError> {
+        let origin = self
+            .origin
+            .clone()
+            .unwrap_or_else(|| "<memory>".to_string());
+        let mut artifact = Artifact::with_origin(origin);
+
+        match format {
+            Format::SingleExport => {
+                self.fill_artifact(&mut artifact, serde_json::from_str(json)?)?
+            }
+            Format::MultiExport => {
+                self.fill_artifact_multi(&mut artifact, serde_json::from_str(json)?)?
+            }
+        }
+
+        Ok(artifact)
+    }
+
+    /// Loads a hardhat artifact from JSON value.
+    pub fn load_from_json(&self, format: Format, value: Value) -> Result<Artifact, ArtifactError> {
+        let origin = self
+            .origin
+            .clone()
+            .unwrap_or_else(|| "<memory>".to_string());
+        let mut artifact = Artifact::with_origin(origin);
+
+        match format {
+            Format::SingleExport => {
+                self.fill_artifact(&mut artifact, serde_json::from_value(value)?)?
+            }
+            Format::MultiExport => {
+                self.fill_artifact_multi(&mut artifact, serde_json::from_value(value)?)?
+            }
+        }
+
+        Ok(artifact)
+    }
+
+    /// Loads a hardhat artifact from disk.
+    pub fn load_from_file(&self, format: Format, path: &Path) -> Result<Artifact, ArtifactError> {
+        let origin = self
+            .origin
+            .clone()
+            .unwrap_or_else(|| path.display().to_string());
+        let mut artifact = Artifact::with_origin(origin);
+
+        let file = File::open(path)?;
```

Comment:
BufReader is usually faster and recommended by serde_json https://docs.serde.rs/serde_json/fn.from_reader.html

```suggestion
        let file = File::open(path)?;
        let reader = std::io::BufReader::new(file);
```